crates/extensions/c8y_http_proxy/src/actor.rs

```diff
@@ -186,56 +197,156 @@ impl C8YHttpProxyActor {
         &mut self,
         device_id: Option<&str>,
     ) -> Result<String, C8YRestError> {
-        let url_get_id = self.end_point.get_url_for_get_id(device_id);
+        let internal_id = if self.cached_identifiers.main_device_internal_id.is_none() {
+            let url_get_id: String = self.end_point.get_url_for_get_id(device_id);
+            let req_builder =
+                |token: String| HttpRequestBuilder::get(&url_get_id).bearer_auth(token);
+
+            self.get_and_set_jwt_token().await?;
+            let request_builder =
+                req_builder(self.cached_identifiers.token.clone().unwrap_or_default());
```

Comment:
As the closure is used along its definition, this seems to be useless.

Reply:
Removed `self.get_and_set_jwt_token()`